source/tyr/tyr/resources.py

```diff
@@ -138,6 +138,31 @@ def post(self, instance_name):
 
         return {'message': 'OK'}, 200
 
+    def delete(self, instance_name=None):
+        parser = reqparse.RequestParser()
+        parser.add_argument('state', type=str, required=False, case_sensitive=True, help='filter by job state')
+        args = parser.parse_args()
+
+        query = models.Job.query
+        if "state" in args and args["state"]:
+            query = query.filter_by(state=args["state"])
+        if instance_name:
+            query = query.join(models.Instance).filter(models.Instance.name == instance_name)
+
+        jobs_to_delete = query.all()
+        if not jobs_to_delete:
+            return json.loads('{}'), 204
+
+        for job in jobs_to_delete:
+            try:
+                db.session.delete(job)
```

Comment:
the job is deleted, do we lose the traceability, too? can we add 'delete' to job's state instead of removing it totally from db?

Reply:
Indeed, traceability is lost. We can add a new value for job state:
https://github.com/CanalTP/navitia/blob/60d49021c1f28a16eacb7f11df939ea049c954e8/source/navitiacommon/navitiacommon/models/__init__.py#L749

Reply:
**To Be Discussed:**
Changing the job state to "deleted" is only relevant for "running" job. "done" and "failed" jobs are finished jobs and won't be taken into account anymore.
In this case, adding an enum might not be relevant, the job "running" can be set to "failed"

Also, to respect CRUD-Style, I think that DELETE should be available for this endpoint.